### PR TITLE
Improve ImagesByProductVariantIdLoader

### PR DESCRIPTION
**Pull request by d-wysocki**
I want to merge this change because...I want to solve performance issue related with getting images from variant.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Review by maarcingebala — APPROVED** [review]

File: saleor/core/utils/random_data.py [hunk 1]
```diff
@@ -293,6 +294,9 @@ def create_product_variants(variants_data):
             product = variant.product
             product.default_variant = variant
             product.save(update_fields=["default_variant", "updated_at"])
+        if create_images:
+            image = variant.product.images.filter().first()
```
[the remark this answers is not in the document]
Comment (korycins) [reply saleor/core/utils/random_data.py:298]: Product has method `get_first_image` is it not the same thing that you are fetching here?

File: saleor/graphql/product/dataloaders/products.py [hunk 2]
```diff
@@ -281,15 +281,38 @@ def batch_load_channel(
...
-        variant_images = VariantImage.objects.filter(variant_id__in=keys)
-        image_map = defaultdict(list)
-        for variant_image in variant_images:
-            image_map[variant_image.variant_id].append(variant_image.image)
-        return [image_map[product_id] for product_id in keys]
+        variant_images = VariantImage.objects.filter(variant_id__in=keys).values_list(
+            "variant_id", "image_id"
+        )
+
+        variant_image_pairs = defaultdict(list)
```
Comment (korycins) [inline saleor/graphql/product/dataloaders/products.py:300]: Can you add the benchmark test to confirm that we solved the issue with many db queries for images?

File: saleor/graphql/product/dataloaders/products.py [hunk 3]
```diff
@@ -281,15 +281,38 @@ def batch_load_channel(
...
+        for variant, image in variant_images:
+            variant_image_pairs[variant].append(image)
```
Comment (IKarbowiak) [inline saleor/graphql/product/dataloaders/products.py:302]: Small change, but it'll improve readability:
```suggestion
        for variant_id, image_id in variant_images:
            variant_image_pairs[variant_id].append(image_id)
```

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/product/dataloaders/products.py (hunk 2 above)
Comment (d-wysocki) [reply saleor/graphql/product/dataloaders/products.py:300]: I updated existing benchmark test related to this changes, which shows that we don't have duplicate queries.

**Pull request merged**